commit b52a4a5f285df49399fe6deefa1bf63dc88cd3d1
Author: Andrew Dunstan <andrew@dunslane.net>
Date:   Mon Apr 7 12:22:41 2025 -0400

    Clean up error messages from 1495eff7bdb
    
    Quote file names, and mostly avoid hard coded file names. Along the way
    make a few other minor improvements.
    
    Discussion: https://postgr.es/m/20250407.152721.1397761902317499205.horikyota.ntt@gmail.com

diff --git a/src/bin/pg_dump/pg_dumpall.c b/src/bin/pg_dump/pg_dumpall.c
index cdb8d84f064..18b544b0214 100644
--- a/src/bin/pg_dump/pg_dumpall.c
+++ b/src/bin/pg_dump/pg_dumpall.c
@@ -521,11 +521,11 @@ main(int argc, char *argv[])
 
 		snprintf(global_path, MAXPGPATH, "%s/global.dat", filename);
 
 		OPF = fopen(global_path, PG_BINARY_W);
 		if (!OPF)
-			pg_fatal("could not open global.dat file: %s", strerror(errno));
+			pg_fatal("could not open \"%s\": %m", global_path);
 	}
 	else if (filename)
 	{
 		OPF = fopen(filename, PG_BINARY_W);
 		if (!OPF)
diff --git a/src/bin/pg_dump/pg_restore.c b/src/bin/pg_dump/pg_restore.c
index fe33b283a01..06c28ab3149 100644
--- a/src/bin/pg_dump/pg_restore.c
+++ b/src/bin/pg_dump/pg_restore.c
@@ -1033,21 +1033,21 @@ get_dbname_oid_list_from_mfile(const char *dumpdirpath, SimpleOidStringList *dbn
 	 * If there is only global.dat file in dump, then return from here as
 	 * there is no database to restore.
 	 */
 	if (!file_exists_in_directory(dumpdirpath, "map.dat"))
 	{
-		pg_log_info("databases restoring is skipped as map.dat file is not present in \"%s\"", dumpdirpath);
+		pg_log_info("database restoring is skipped as \"map.dat\" is not present in \"%s\"", dumpdirpath);
 		return 0;
 	}
 
 	snprintf(map_file_path, MAXPGPATH, "%s/map.dat", dumpdirpath);
 
 	/* Open map.dat file. */
 	pfile = fopen(map_file_path, PG_BINARY_R);
 
 	if (pfile == NULL)
-		pg_fatal("could not open map.dat file: \"%s\"", map_file_path);
+		pg_fatal("could not open \"%s\": %m", map_file_path);
 
 	/* Append all the dbname/db_oid combinations to the list. */
 	while ((fgets(line, MAXPGPATH, pfile)) != NULL)
 	{
 		Oid			db_oid = InvalidOid;
@@ -1062,15 +1062,17 @@ get_dbname_oid_list_from_mfile(const char *dumpdirpath, SimpleOidStringList *dbn
 		dbname = line + strlen(db_oid_str) + 1;
 
 		/* Remove \n from dbname. */
 		dbname[strlen(dbname) - 1] = '\0';
 
-		pg_log_info("found database \"%s\" (OID: %u) in map.dat file while restoring.", dbname, db_oid);
+		pg_log_info("found database \"%s\" (OID: %u) in \"%s\"",
+					dbname, db_oid, map_file_path);
 
 		/* Report error and exit if the file has any corrupted data. */
 		if (!OidIsValid(db_oid) || strlen(dbname) == 0)
-			pg_fatal("invalid entry in map.dat file at line : %d", count + 1);
+			pg_fatal("invalid entry in \"%s\" at line : %d", map_file_path,
+					 count + 1);
 
 		simple_oid_string_list_append(dbname_oid_list, db_oid, dbname);
 		count++;
 	}
 
@@ -1114,11 +1116,11 @@ restore_all_databases(PGconn *conn, const char *dumpdirpath,
 
 	/* If map.dat has no entry, return after processing global.dat */
 	if (dbname_oid_list.head == NULL)
 		return process_global_sql_commands(conn, dumpdirpath, opts->filename);
 
-	pg_log_info("found total %d database names in map.dat file", num_total_db);
+	pg_log_info("found %d database names in \"map.dat\"", num_total_db);
 
 	if (!conn)
 	{
 		pg_log_info("trying to connect database \"postgres\"");
 
@@ -1286,11 +1288,11 @@ process_global_sql_commands(PGconn *conn, const char *dumpdirpath, const char *o
 
 	/* Open global.dat file. */
 	pfile = fopen(global_file_path, PG_BINARY_R);
 
 	if (pfile == NULL)
-		pg_fatal("could not open global.dat file: \"%s\"", global_file_path);
+		pg_fatal("could not open \"%s\": %m", global_file_path);
 
 	/*
 	 * If outfile is given, then just copy all global.dat file data into
 	 * outfile.
 	 */
@@ -1333,11 +1335,11 @@ process_global_sql_commands(PGconn *conn, const char *dumpdirpath, const char *o
 		PQclear(result);
 	}
 
 	/* Print a summary of ignored errors during global.dat. */
 	if (n_errors)
-		pg_log_warning("errors ignored on global.dat file restore: %d", n_errors);
+		pg_log_warning("ignored %d errors in \"%s\"", n_errors, global_file_path);
 
 	fclose(pfile);
 
 	return n_errors;
 }
